### PR TITLE
Fix #108 Add logging for errors handling large server responses

### LINKED ISSUE #108: Add handling for too large HTTP responses
We should improve error handling in case that HTTP response from server is too large. In this case the status code is not known, so logging of applications which use the REST client is not sufficient.

Recommended steps to fix the issue:
- prepare a REST endpoint with HTTP POST method which returns a response 10 MB large
- prepare a sample code in which REST client is used to call this large endpoint
- catch the exception from WebClient for too large response and log it using SLF4J on `WARN` level and test this manually using sample code
- write a unit test for this scenario and add it into lime-java-core library

### PR DESCRIPTION
**Pull request by attopp**
- Add logging to DefaultRestClient for blocking post
 - Add sample REST endpoint to PublicTestRestController
 - Add test case into DefaultRestClientTest to verify behaviour

### CODE REVIEW

File: rest-client-base/src/test/java/com/wultra/core/rest/client/base/DefaultRestClientTest.java [hunk 1]
```diff
@@ -657,6 +662,23 @@ void testPostWithMultipartData() throws RestClientException {
         assertEquals(requestData, responseEntity.getBody().getResponseObject().getResponse());
     }
 
+    @Test
+    void testPostWithLargeServerResponse() {
+        Logger defaultRestClientLogger = (Logger) LoggerFactory.getLogger(DefaultRestClient.class);
```
Comment (romanstrobl) [inline rest-client-base/src/test/java/com/wultra/core/rest/client/base/DefaultRestClientTest.java:667]: Can we add the `final` modifier to the fields which are immutable?

File: rest-client-base/src/test/java/com/wultra/core/rest/client/base/DefaultRestClientTest.java [hunk 2]
```diff
@@ -657,6 +662,23 @@ void testPostWithMultipartData() throws RestClientException {
...
+        listAppender.start();
+        defaultRestClientLogger.addAppender(listAppender);
+
+        final RestClientException exception = assertThrows(RestClientException.class,
+                () -> restClient.post("/object-response-large", null, new ParameterizedTypeReference<Response>() {
+                }));
+
+        List<ILoggingEvent> logsList = listAppender.list;
+        assertFalse(logsList.isEmpty());
+        assertEquals(1, logsList.stream().filter(logEvent -> logEvent.getMessage().equals("Error while retrieving large server response.")).count());
```
Comment (romanstrobl) [inline rest-client-base/src/test/java/com/wultra/core/rest/client/base/DefaultRestClientTest.java:678]: This is a really long line, can we split the statements?

**Review by romanstrobl — APPROVED** [review]
I have two minor comments, otherwise good job!

Before merging a contributor agreement should be signed.

**Comment by banterCZ** [comment]
@attopp Sorry, we do not have the _contributing_ file yet, but mind that in order to eliminate the risk that someone unknown smuggles an unknown commit to our GitHub repositories, **we require all developers to sign all their commits**.
All developers can then prove their identity via the signature and the commits will be verified.
[Learn more about signing commits](https://docs.github.com/articles/about-gpg/).

**Comment by jandusil** [comment]
I am closing the issue as I am now a proper contributor of the repo under a new account. The same issue is dealt with in a new PR https://github.com/wultra/lime-java-core/pull/196

**Pull request closed without merging**